cpp/src/arrow/CMakeLists.txt

```diff
@@ -301,13 +301,24 @@ arrow_add_pkg_config("arrow")
 #
 
 add_arrow_test(allocator-test)
-add_arrow_test(array-test
-               SOURCES
-               array-test.cc
-               array-binary-test.cc
-               array-dict-test.cc
-               array-list-test.cc
-               array-struct-test.cc)
+
+if(WIN32)
```

Comment:
@javierluraschi @kou In the future, please add a comment when you add a special case like this, so that we know why it is there.
(also, this test should really be more narrow)

Reply:
OK. Sorry for this.
(I should have pushed a change for https://github.com/apache/arrow/pull/3693#discussion_r261481746 by myself.)